executable-spec/test/Cardano/Ledger/Spec/STS/Chain/Chain/Properties.hs

```diff
@@ -46,3 +53,231 @@ qc_revealsAreClassified
       & dom
       & length
       & fromIntegral
+
+-- | Test coverage describes the test space that we are targeting for.
+-- Describe the most essential characteristics of our generated traces.
+--
+-- TODO: this must be updated to reflect all phases in the lifecycle of
+-- a software update and not just the Ideation phase.
+relevantCasesAreCovered :: QC.Property
+relevantCasesAreCovered
+  = QC.withMaxSuccess 300
+  $ STS.Gen.forAllTrace @(CHAIN ShortHash) @() maxTraceLength ()
+  $ \traceSample ->
+        QC.checkCoverage $
+          -- traces should be long enough to allow for the stabilization
+          -- of events
+          QC.cover 80
+            ( (fromIntegral $ Trace.traceLength traceSample)
+              >=
+              getMinTraceLength  (Chain.k $ Trace._traceEnv traceSample)
+            )
+            "Trace length is long enough to allow for the stabilization of events"
+            $
+
+          -- 80% of traces should include a 20% percent of update payload
+          QC.cover 80
+            ( (updatePayloadPct traceSample) >= 15
+            &&
+              (updatePayloadPct traceSample) <= 25
+            )
+            "a reasonable pct of update payload appears in the trace "
+            $
+
+          -- Lifecycle coverage:
+          -- There is at least one proposal in every phase of the lifecycle
+          QC.cover 80
+            (lifecycleCoverage traceSample)
+            "The lifecycle of a software update is sufficently covered"
+            $
+
+          -- X% of traces should: there are SIPs that got approved
+          QC.cover 25
```

Comment:
I wonder if we should check that the last state contains a certain outcome vs any state contains a certain outcome. The problem with checking the last state is that we're missing out outcome events that happen during trace execution.

Reply:
How can we miss outcome events, they are all stored in the vresips state, without deletion?
The only case is when we have a revoting, but that will lead also to some outcome.

Moreover, in early states there are might not be so many voting results yet.

Reply:
But don't we overwrite the `vresips`? So you might have a rejection that in the final state is turned into an approval and miss to register this, don't you?